model/vfs/vfsswift/swift.go

```diff
@@ -0,0 +1,70 @@
+package vfsswift
+
+import (
+	"github.com/cozy/swift"
+	multierror "github.com/hashicorp/go-multierror"
+)
+
+// maxNbFilesToDelete is the maximal number of files that we will try to delete
+// in a single call to swift.
+const maxNbFilesToDelete = 8000
+
+// maxSimultaneousCalls is the maximal number of simultaneous calls to Swift to
+// delete files in the same container.
+const maxSimultaneousCalls = 8
+
+func deleteContainer(c *swift.Connection, container string) error {
+	_, _, err := c.Container(container)
+	if err == swift.ContainerNotFound {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	objectNames, err := c.ObjectNamesAll(container, nil)
+	if err != nil {
+		return err
+	}
+	if len(objectNames) > 0 {
+		if err = deleteContainerFiles(c, container, objectNames); err != nil {
+			return err
+		}
+	}
+	return c.ContainerDelete(container)
+}
+
+func deleteContainerFiles(c *swift.Connection, container string, objectNames []string) error {
+	nb := 1 + (len(objectNames)-1)/maxNbFilesToDelete
+	ch := make(chan error)
+
+	// Use a system of tokens to limit the number of simultaneous calls to
+	// Swift: only a goroutine that has a token can make a call.
+	tokens := make(chan int, maxSimultaneousCalls)
+	for k := 0; k < maxSimultaneousCalls; k++ {
+		tokens <- k
+	}
+
+	for i := 0; i < nb; i++ {
+		begin := i * maxNbFilesToDelete
+		end := (i + 1) * maxNbFilesToDelete
+		objectToDelete := objectNames[begin:end]
+		go func() {
+			k := <-tokens
+			_, err := c.BulkDelete(container, objectToDelete)
```

Comment:
I think you need to check if your container is a versionned container and if this is the case, to delete all (swift) versions for each file.

Reply:
In swift layout v3, we don't have versioned containers. In swift layout v2, it is not done like this. When we delete a versioned container, we disable the versioning, then we call this function to delete the container with the versions, and finally, we call again this function to delete the container with the files. I think it is done like this as it is a lot faster, that trying to delete version after version.